sd-core/src/language/llvm_ir.rs

```diff
@@ -0,0 +1,670 @@
+use derive_more::{Display, From};
+use either::Either;
+use itertools::Itertools;
+use llvm_ir::{
+    instruction::{InlineAssembly, RMWBinOp},
+    module::{GlobalAlias, GlobalIFunc, GlobalVariable},
+    BasicBlock, Constant, ConstantRef, FPPredicate, Function, Instruction, IntPredicate, Module,
+    Name, Operand, Terminator,
+};
+use tracing::debug;
+
+use super::{Fresh, Language, OpInfo, CF};
+use crate::{
+    common::Matchable,
+    hypergraph::traits::{WireType, WithType},
+};
+
+#[derive(Copy, Clone, Eq, PartialEq, Hash, Debug)]
+pub struct LlvmIrSettings {
+    pub sym_name_linking: bool,
+}
+
+impl Default for LlvmIrSettings {
+    fn default() -> Self {
+        Self {
+            sym_name_linking: true,
+        }
+    }
+}
+
+pub struct LlvmIr;
+
+pub type Expr = super::Expr<LlvmIr>;
+pub type Bind = super::Bind<LlvmIr>;
+pub type Value = super::Value<LlvmIr>;
+pub type Thunk = super::Thunk<LlvmIr>;
+pub type Block = super::Block<LlvmIr>;
+
+impl Language for LlvmIr {
+    type Op = Op;
+    type Var = Var;
+    type Addr = String;
+    type VarDef = Var;
+    type BlockAddr = Name;
+    type Symbol = Box<String>;
+}
+
+#[derive(Clone, PartialEq, Debug, Hash, Display, From)]
+pub enum Op {
+    // constant-like
```

Comment:
Any reason these aren't docstrings?

Reply:
It's more like a code comment than actual documentation